mne/viz/_brain/_timeviewer.py

```diff
@@ -355,7 +357,21 @@ def __init__(self, brain, show_traces=False):
 
         # show everything at the end
         self.toggle_interface()
-        self.brain.show()
+        with self.ensure_minimum_sizes():
+            self.brain.show()
+
+    @contextlib.contextmanager
+    def ensure_minimum_sizes(self):
+        sz = self.brain._size
+        adjust_mpl = self.show_traces and not self.separate_canvas
+        if adjust_mpl:
+            self.mpl_canvas.canvas.setMinimumSize(
+                sz[0], int(round(sz[1] / self.interactor_stretch)))
+        try:
+            yield
+        finally:
+            if adjust_mpl:
+                self.mpl_canvas.canvas.setMinimumSize(0, 0)
```

Comment:
This flow control seems cleaner to me, WDYT?
```suggestion
        if not adjust_mpl:
            yield
        else:
            self.mpl_canvas.canvas.setMinimumSize(
                sz[0], int(round(sz[1] / self.interactor_stretch)))
            try:
                yield
            finally:
                self.mpl_canvas.canvas.setMinimumSize(0, 0)
```

Reply:
One if to rule them all